### PR TITLE
6724 - Fixed UI issues and changed FED to not load data when resuming.

### PR DESCRIPTION
**Pull request by andrepestana-aot**
*Issue #:* /bcgov/entity#6724

*Description of changes:*

- Fixed UI issues
- Changed FED to not load data when resuming
- Fixed tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

File: src/components/Summary/AlterationSummary.vue [hunk 1]
```diff
@@ -162,7 +162,11 @@
               @valid="setEffectiveDateValid($event)"
             />
 
-            <v-card flat class="px-16 pb-8 mt-n12" v-if="isFutureEffective && isEffectiveDateTimeValid">
+            <v-card
+              flat
+              class="px-16 pb-8 mt-n12"
+              id="effectiveness-text"
```
Comment (severinbeauvais) [inline src/components/Summary/AlterationSummary.vue:168]: I hate to pick on an id name, but this doesn't mean what you think it means.

File: src/components/common/EffectiveDateTime.vue [hunk 2]
```diff
@@ -221,7 +221,8 @@ export default class EffectiveDateTime extends Mixins(DateMixin) {
 
   /** Called when component is mounted. */
   mounted (): void {
-    this.parseInitialEffectiveDateTime()
+    /** It was decided not doing it for now */
```
Comment (severinbeauvais) [inline src/components/common/EffectiveDateTime.vue:224]: Confirmed by Linda?

File: src/components/common/EffectiveDateTime.vue [hunk 3]
```diff
@@ -462,4 +465,18 @@ export default class EffectiveDateTime extends Mixins(DateMixin) {
     color: $BCgovInputError !important;
   }
 }
+::v-deep {
+  .date-picker-form  .v-input__control > .v-input__slot:before {
```
Comment (severinbeauvais) [inline src/components/common/EffectiveDateTime.vue:469]: This element is not in this component. Should this be moved to the date picker component instead?

Comment (severinbeauvais) [reply src/components/common/EffectiveDateTime.vue:469]: Same for the v-icon below I think.

File: src/components/common/EffectiveDateTime.vue [hunk 4]
```diff
@@ -462,4 +465,18 @@ export default class EffectiveDateTime extends Mixins(DateMixin) {
...
+    border-image: none;
+  }
+  .theme--light.v-icon.v-icon.v-icon--disabled {
+    color: $app-blue !important;
+  }
+  .theme--light.v-input--is-disabled {
```
Comment (severinbeauvais) [inline src/components/common/EffectiveDateTime.vue:475]: I'm not sure if this rule applies to this component's v-inputs (if any?) or the ones in the date picker component, but if the latter then this probably shouldn't be here. (Also line 478.)

File: tests/unit/EffectiveDateTime.spec.ts [hunk 5]
```diff
@@ -55,7 +56,7 @@ describe('Effective Date Time component', () => {
     valid: false,
     isFutureEffective: true,
     // effective date is more than 10 days from now
```
Comment (severinbeauvais) [inline tests/unit/EffectiveDateTime.spec.ts:58]: Move comment up please (so it makes sense).

File: src/components/common/EffectiveDateTime.vue (hunk 2 above)
Comment (andrepestana-aot) [reply src/components/common/EffectiveDateTime.vue:224]: Yes, it was.

File: tests/unit/EffectiveDateTime.spec.ts (hunk 5 above)
Comment (andrepestana-aot) [reply tests/unit/EffectiveDateTime.spec.ts:58]: 👍

File: src/components/common/EffectiveDateTime.vue (hunk 3 above)
Comment (andrepestana-aot) [reply src/components/common/EffectiveDateTime.vue:469]: ok. Will change the DatePicker.

File: src/components/Summary/AlterationSummary.vue (hunk 1 above)
Comment (andrepestana-aot) [reply src/components/Summary/AlterationSummary.vue:168]: So what does it really mean?

Comment (severinbeauvais) [reply src/components/Summary/AlterationSummary.vue:168]: "Effectiveness" is how effective something is.

"Effective date" is the date it is effective on.

Hope you're ok with the joke above :)

Comment (andrepestana-aot) [reply src/components/Summary/AlterationSummary.vue]: sure! I like jokes (when I get them 🤣)

File: src/components/common/EffectiveDateTime.vue (hunk 4 above)
Comment (andrepestana-aot) [reply src/components/common/EffectiveDateTime.vue]: Fixed. Now DatePicker has been changed.

File: tests/unit/EffectiveDateTime.spec.ts (hunk 5 above)
Comment (andrepestana-aot) [reply tests/unit/EffectiveDateTime.spec.ts]: Done now.

File: src/components/Summary/AlterationSummary.vue (hunk 1 above)
Comment (andrepestana-aot) [reply src/components/Summary/AlterationSummary.vue]: I changed the id though...

**Comment by severinbeauvais** [comment]
Sorry about the conflicts :/

Also, please update app version to 0.6.7 now.

**Review by severinbeauvais — APPROVED** [review]

**Comment by severinbeauvais** [comment]
@argush3 Does this PR look OK to you?

**Comment by argush3** [comment]
> @argush3 Does this PR look OK to you?

Going through it right now.

**Comment by argush3** [comment]
Just the one question but otherwise looks good to me. 👍

File: src/components/common/EffectiveDateTime.vue [hunk 6]
```diff
@@ -268,7 +269,7 @@ export default class EffectiveDateTime extends Mixins(DateMixin) {
 
     const isDateValid = this.$refs.datePickerRef.validateForm()
     const isTimeValid = this.$refs.form.validate()
-    if (isDateValid && isTimeValid) {
+    if (isDateValid && isTimeValid && !!this.selectHour.length && !!this.selectMinute.length) {
```
[the remark this answers is not in the document]
Comment (andrepestana-aot) [reply src/components/common/EffectiveDateTime.vue:272]: I changed the way the EffectiveDateTime component to allow selecting the date, hour, minute or period at any time. Before it had an order: first the date, then hour and minute.

`isDateValid` and `isTimeValid` return true when the form is valid. That means that when we don't have a selected hour or minute, that could return true. So doing all this checks everything works fine now.

File: tests/unit/EffectiveDateTime.spec.ts [hunk 7]
```diff
@@ -31,6 +31,9 @@ describe('Effective Date Time component', () => {
   let wrapperFactory: any
   const today = new Date()
 
+  // invalid future effective date is more than 10 days from now
+  const futureInvalidDate = new Date(today.getTime() + 11 * 24 * 60 * 60 * 1000)
```
[the remark this answers is not in the document]
Comment (andrepestana-aot) [reply tests/unit/EffectiveDateTime.spec.ts:35]: 👍

File: src/components/common/EffectiveDateTime.vue (hunk 6 above)
[the remark this answers is not in the document]
Comment (argush3) [reply src/components/common/EffectiveDateTime.vue:272]: ok, makes sense.  thanks for the explanation

**Review by argush3 — APPROVED** [review]

**Pull request merged**